src/zarr/v3/group.py

```diff
@@ -271,8 +289,32 @@ def __repr__(self):
     async def nchildren(self) -> int:
         raise NotImplementedError
 
-    async def children(self) -> AsyncIterator[AsyncArray, AsyncGroup]:
-        raise NotImplementedError
+    async def children(self) -> AsyncGenerator[AsyncArray, AsyncGroup]:
+        """
+        Returns an AsyncGenerator over the arrays and groups contained in this group.
+        This method requires that `store_path.store` supports directory listing.
+        """
+        if not self.store_path.store.supports_listing:
+            msg = (
+                f"The store associated with this group ({type(self.store_path.store)}) "
+                "does not support listing, "
+                "specifically via the `list_dir` method. "
+                "This function requires a store that supports listing."
+            )
+
+            raise ValueError(msg)
+        subkeys = await self.store_path.store.list_dir(self.store_path.path)
```

Comment:
I think the ideal async way to do this would be for `list_dir` to return an `AsyncGenerator`. Then your for loop below would look something like:

```python
async for key in subkeys:
    # filter
    if key in ("zarr.json", ".zgroup", ".zattrs"):
         ...
         yield await self.getitem(subkey)
         
     ...
```